hail/src/main/scala/is/hail/expr/types/physical/PSubsetStruct.scala

```diff
@@ -0,0 +1,214 @@
+
+  override def loadField(structAddress: Long, fieldIdx: Int): Long =
+    ps.loadField(structAddress, idxMap(fieldIdx))
+
+  override def loadField(structAddress: Code[Long], fieldIdx: Int): Code[Long] =
+    ps.loadField(structAddress, idxMap(fieldIdx))
+
+  override def setFieldPresent(structAddress: Code[Long], fieldName: String): Code[Unit] = ???
+
+  override def setFieldMissing(structAddress: Code[Long], fieldName: String): Code[Unit] = ???
+
+  override def setFieldMissing(structAddress: Long, fieldIdx: Int): Unit = ???
+
+  override def setFieldMissing(structAddress: Code[Long], fieldIdx: Int): Code[Unit] = ???
+
+  override def setFieldPresent(structAddress: Long, fieldIdx: Int): Unit = ???
+
+  override def setFieldPresent(structAddress: Code[Long], fieldIdx: Int): Code[Unit] = ???
+
+  override def appendKey(key: String, sig: PType): PStruct = ???
+
+  def insertFields(fieldsToInsert: TraversableOnce[(String, PType)]): PSubsetStruct = ???
+
+  override def initialize(structAddress: Long, setMissing: Boolean): Unit = ???
+
+  override def stagedInitialize(structAddress: Code[Long], setMissing: Boolean): Code[Unit] = ???
+
+  override def allocate(region: Region): Long = ???
+
+  override def allocate(region: Code[Region]): Code[Long] = ???
+
+  override def setRequired(required: Boolean): PType = ???
+
+  override def copyFromType(mb: EmitMethodBuilder[_], region: Value[Region], srcPType: PType, srcAddress: Code[Long], deepCopy: Boolean): Code[Long] = ???
+
+  override def copyFromTypeAndStackValue(mb: EmitMethodBuilder[_], region: Value[Region], srcPType: PType, stackValue: Code[_], deepCopy: Boolean): Code[_] = ???
+
+  override protected def _copyFromAddress(region: Region, srcPType: PType, srcAddress: Long, deepCopy: Boolean): Long = ???
+
+  override def constructAtAddress(mb: EmitMethodBuilder[_], addr: Code[Long], region: Value[Region], srcPType: PType, srcAddress: Code[Long], deepCopy: Boolean): Code[Unit] = ???
+
+  override def constructAtAddress(addr: Long, region: Region, srcPType: PType, srcAddress: Long, deepCopy: Boolean): Unit = ???
+}
+
+object PSubsetStructSettable {
+  def apply(cb: EmitCodeBuilder, pt: PStruct, name: String, sb: SettableBuilder): PSubsetStructSettable = {
+    new PSubsetStructSettable(pt, sb.newSettable(name))
+  }
+}
+
+class PSubsetStructSettable(val pt: PStruct, val a: Settable[Long]) extends PBaseStructValue with PSettable {
+  def get: PSubsetStructCode = new PSubsetStructCode(pt, a)
+
+  // Again, is fieldIdx here the backing PStruct?
+  def loadField(cb: EmitCodeBuilder, fieldIdx: Int): IEmitCode = {
+    IEmitCode(cb,
+      pt.isFieldMissing(a, fieldIdx),
+      pt.fields(fieldIdx).typ.load(pt.fieldOffset(a, fieldIdx)))
+  }
+
+  def store(pv: PCode): Code[Unit] = {
+    a := pv.asInstanceOf[PSubsetStructCode].a
+  }
+}
+
+// TODO: We don't allow construction, so what do we do here for store?
+class PSubsetStructCode(val pt: PStruct, val a: Code[Long]) extends PBaseStructCode {
```

Comment:
Could you help me understand this and the above Value?

I thought that we were to encapsulate (PType, Value). Here we're not doing that. I could understand the logic in the sense that the Code/Value refers to a constructible.  At the same time, it seems strange to, for instance, loadField where fieldIdx refers to the underlying struct and not the PSubsetStruct field idx. Similarly, it may be strange to `store` ins PSubsetStructCode and end up constructing/materializing the full underlying struct.

Reply:
Code/Value are about usage, not construction. We can absolutely create a value for a PSubsetStruct.

Reply:
But that value isn't a PSubsetStruct?

Reply:
I can't find any usages of `store`. It seems like this can put something in a region?

I think the same comment applies as above -- PSubsetStructCode should take a PBaseStructCode and a PSubsetStruct to construct.